### PR TITLE
HACK BUG FIX: replace fancy memory management

### PR DESCRIPTION
**Pull request by daniel-noland**
The existing memory management algorithm seems to leak memory very badly.  This is a temp replacement with a much simpler algorithm to trigger cleanup.

Note that you must disable inMemoryBackup for this to work properly (I'm not actually sure why at this time).

### CODE REVIEW

**Review by OferMania — APPROVED** [review]

File: src/LogCleaner.cc [hunk 1]
```diff
@@ -789,13 +789,16 @@ LogCleaner::Balancer::compactionFailed()
...
+    ++count;
+    if (count % 200 == 0) {
         return CLEAN_DISK;
-
-    if (!cleaner->disableInMemoryCleaning && isMemoryLow(thread))
-        return COMPACT_MEMORY;
-
-    return SLEEP;
+    }
+    return count % 25 == 0 ? COMPACT_MEMORY : SLEEP;
```
Comment (james-munson) [inline src/LogCleaner.cc:801]: How often is requestTask() called?  Not worried about wraparound of a 64-bit counter, just wondering what that translates to for kicking compaction or cleaning.

Comment (adickson66) [reply src/LogCleaner.cc:801]: I have same question, thanks.

**Review by adickson66 — APPROVED** [review]
The only question I have is the same as James's on the frequency. 
Approving with the assumption there is good justification for the 200 and 25 numbers, just want to know.

File: src/LogCleaner.cc (hunk 1 above)
Comment (daniel-noland) [reply src/LogCleaner.cc:801]: That is a fair point.  I thought about it but, assuming we increment 5 billion times per second then:

```python3
 >>> 2**64 / (5*10**9) / (24 * 60 * 60 * 365.25)
116.90840921812529
```

it takes 116.9 years for us to wrap our counter.  Likely not worth messing with when we plan to rip out the change as soon as possible.

**Review by james-munson — APPROVED** [review]

**Comment by daniel-noland** [comment]
> The only question I have is the same as James's on the frequency.
> Approving with the assumption there is good justification for the 200 and 25 numbers, just want to know.

I don't  have a first principals justification really.  Just tuned it to behave reasonably on my machine.  We can adjust but I really don't want this "fix" to be long for the world anyway.

**Pull request merged**